src/CMakeLists.txt

```diff
@@ -167,24 +167,34 @@ ADD_FEATURE_INFO(threadsafe THREADSAFE "threadsafe support")
 ### Detect and link against dependencies
 ###
 
+FUNCTION(OBJECT_LIBRARY_LINK TARGET)
```

Comment:
I don't like the `OBJECT_LIBRARY_LINK` name, as we already have too much `TARGET` `LINK` and `LIBRARY` around (after just having read about the new `TARGET_LINK_OBJECTS` 🤣). I'd recommend sticking closer to the known `TARGET_LINK_LIBRARIES`, so `LG2_TARGET_LINK_LIBRARIES` ?